### PR TITLE
rfc: add 20230307-stage-with-schema.md

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

**Comment by youngsofun** [comment]
https://github.com/datafuselabs/databend/issues/created_by/youngsofun

File: docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+)
+```
+
+if the same schema is used frequently, user can create table for it and use table name.
+
+```sql
+SELECT  c1, c2 + c1, trim(c2)  FROM @my_stage(schema='db_name.table_name')
+```
+
+and user can get knowledge about the data with `desc <table>` instead of `infer schema` each time.
```
Comment (Xuanwo) [inline docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md:42]: Does this mean that if a schema is not attached, we will fall back to inferring the schema instead?

File: docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## Reference-level explanation
+
+```sql
+
```
Comment (Xuanwo) [inline docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md:47]:
```suggestion
```

File: docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+SELECT <exprs> FROM @<stage>/'<uri>'(SCHEMA= (<schema> | "<table_name>"), ..)
```
Comment (Xuanwo) [inline docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md:48]:
```sql
select * from 's3://bucket'(SCHEMA="my_table")
```

The use of this URI appears somewhat unusual.

File: docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md (hunk 1 above)
Comment (youngsofun) [reply docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md:42]: yes.

File: docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md (hunk 3 above)
Comment (youngsofun) [reply docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md:47]: sorry for typo, "@<stage>|'<uri>" not `<stage>/'<uri>`

File: docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md (hunk 2 above)
Comment (youngsofun) [reply docs/doc/90-contributing/01-rfcs/20230307-stage-with-schema.md]: fixed, thanks!

**Pull request merged**